shims/net/index.ts

```diff
@@ -62,17 +62,21 @@ export interface SocketDefaults {
 }
 type GlobalOnlyDefaults = 'poolQueryViaFetch' | 'fetchEndpoint' | 'fetchConnectionCache' | 'fetchFunction';
 
+const transformHost = (host: string): string => {
+  return host.replace(/^[^.]*/, 'api');
+}
```

Comment:
Here and where we set the endpoint option, should we first check that the host has a plausible structure? For example, that it has at least two dots in it, or maybe even that it includes an AWS region? I ask because we advertise the driver as capable of being used with your own DB back-end and proxy, and this may break backwards-compatibility.

Reply:
Thanks for making the change, but I don't know that this works. Some DBs are Vercel Postgres DBs, and will be accessed at e.g. `my-endpoint.eu-central-1.postgres.vercel-storage.com`. Presumbly similar things may happen in future. So maybe just checking for two dots (and calling this a possibly-breaking change) is the best we can do?

Reply:
Our endpoints have a distinctive form of ep-{word}-{word}-{alphanumerics}. Maybe we can rely on that?

Reply:
Good idea!

Reply:
Yes, as long as we're pretty sure that this format (a) won't change any time soon and (b) is true for all partners (including e.g Vercel Storage).

Reply:
yea, the endpoint id format is more or less stable in this sense.